src/TokenForm.js

```diff
@@ -27,7 +27,7 @@ const TokenForm = () => {
       });
       return unsubscribe;
     };
-
+    fetchItems();
```

Comment:
Ah, ok...so it wasn't fully removed, but just not invoked.

So, for the memory leak, what I think that you actually want to do is return `unsubscribe`. However, because `unsubscribe` is initialized inside of a `fetchItems` you'll need to assign it to a variable initialized in the outer body of the function. Typically, for reassignment in React, we use [refs](https://reactjs.org/docs/refs-and-the-dom.html). 
In this case, I'd recommend that rather than returning `fetchItems` at the end, you should do something like:

```javascript
const unsubscribeRef = useRef();

useEffect(() => {
    const fetchItems = async () => {
      const response = collection(db, 'shopping-list');
      const itemList = query(response);
      const unsubscribe = onSnapshot(itemList, (querySnapshot) => {
        const tokens = querySnapshot.docs.reduce((acc, doc) => {
          const { userToken } = doc.data();
          return [...acc, userToken];
        }, []);
        console.log('tokens: ', tokens);
        setTokens(tokens);
      });
      return unsubscribe;
    };
    fetchItems();
    return unsubscribeRef.current;
}, [db]);
```

Reply:
Thank you Aidan for doing the change and thank you, Mike, for telling us how to fix the issue!! Now it is working